lib/ts/recipe/accountlinking/recipe.ts

```diff
@@ -146,13 +152,22 @@ export default class Recipe extends RecipeModule {
             return user;
         }
 
-        // then, we try and find a primary user based on the email / phone number / third party ID.
+        const accountInfo: AccountInfoInput = {
+            ...user.loginMethods[0],
+            webauthn: undefined,
+        };
+        // any credentialId will identify the same user
+        if (user.loginMethods[0]?.webauthn?.credentialIds.length) {
+            accountInfo.webauthn = {
+                credentialId: user.loginMethods[0].webauthn.credentialIds[0],
+            };
+        }
+        // then, we try and find a primary user based on the email / phone number / third party ID / credentialId.
```

Comment:
Does it make sense to search by `credentialId` here? I don't think so: we are looking for a user to link to the current recipe user, but any search using the credentialId of the current user "will identify the same user" which is the current one.

Reply:
Done.